src/MercadoPago/Client/Preference/PreferenceClient.php

```diff
@@ -0,0 +1,103 @@
+    public function __construct()
+    {
+        parent::__construct(MercadoPagoConfig::getHttpClient());
+    }
+
+    /**
+     * Method responsible for creating preference.
+     * @param array $request preference data.
+     * @param mixed $request_options request options to be sent.
+     * @return \MercadoPago\Resources\Preference preference created.
+     */
+    public function create(array $request, ?MPRequestOptions $request_options = null): Preference
+    {
+        try {
+            $response = parent::send(self::$URL, HttpMethod::POST, json_encode($request), null, $request_options);
+            $result = Serializer::deserializeFromJson(Preference::class, $response->getContent());
+            $result->setResponse($response);
+            return $result;
+        } catch (MPApiException | \Exception $e) {
+            throw $e;
+        }
+    }
+
+    /**
+     * Method responsible for getting preference.
+     * @param string $id preference id.
+     * @param mixed $request_options request options to be sent.
+     * @return \MercadoPago\Resources\Preference preference found.
+     */
+    public function get(string $id, ?MPRequestOptions $request_options = null): Preference
+    {
+        try {
+            $response = parent::send(sprintf(self::$URL_WITH_ID, $id), HttpMethod::GET, null, null, $request_options);
+            $result = Serializer::deserializeFromJson(Preference::class, $response->getContent());
+            $result->setResponse($response);
+            return $result;
+        } catch (MPApiException | \Exception $e) {
+            throw $e;
+        }
+    }
+
+    /**
+     * Method responsible for update preference.
+     * @param string $id preference id.
+     * @param array $request preference data.
+     * @param mixed $request_options request options to be sent.
+     * @return \MercadoPago\Resources\Preference preference canceled.
+     */
+    public function update(string $id, array $request, ?MPRequestOptions $request_options = null): Preference
+    {
+        try {
+            $response = parent::send(sprintf(self::$URL_WITH_ID, $id), HttpMethod::PUT, json_encode($request), null, $request_options);
+            $result = Serializer::deserializeFromJson(Preference::class, $response->getContent());
+            $result->setResponse($response);
+            return $result;
+        } catch (MPApiException | \Exception $e) {
+            throw $e;
+        }
+    }
+
+    /**
+     *  Method responsible for search preferences.
+     * @param \MercadoPago\Net\MPSearchRequest $request search request.
+     * @param mixed $requestOptions request options to be sent.
+     * @return \MercadoPago\Resources\PreferenceSearch search results.
+     */
+    public function search(MPSearchRequest $request, ?MPRequestOptions $requestOptions = null): PreferenceSearch
+    {
+        try {
+            $queryParams = isset($request) ? $request->getParameters() : null;
```

Comment:
Em alguns lugares vi que o padrão para definir variáveis é o snake case
```suggestion
            $query_params = isset($request) ? $request->getParameters() : null;
```